cruise-control/src/test/java/com/linkedin/kafka/cruisecontrol/servlet/parameters/ParameterJsonStructureTest.java

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright 2017 LinkedIn Corp. Licensed under the BSD 2-Clause License (the "License"). See License in the project root for license information.
+ */
+package com.linkedin.kafka.cruisecontrol.servlet.parameters;
+
+import com.linkedin.cruisecontrol.servlet.parameters.CruiseControlParameters;
+import io.swagger.v3.oas.models.PathItem;
+import io.swagger.v3.oas.models.Paths;
+import io.swagger.v3.oas.models.OpenAPI;
+import io.swagger.v3.oas.models.parameters.Parameter;
+import io.swagger.v3.parser.OpenAPIV3Parser;
+
+import io.swagger.v3.parser.core.models.ParseOptions;
+import io.swagger.v3.parser.core.models.SwaggerParseResult;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.List;
+import java.util.Set;
+import java.util.Map;
+import java.util.HashMap;
+import java.util.TreeSet;
+
+public class ParameterJsonStructureTest {
+  private Map<String, Class> _endpointToClass;
+  OpenAPI _openAPI;
+  /**
+   * Specify endpoints to be tested
+   */
+  @Before
+  public void setupParameterClasses() {
+    _endpointToClass = new HashMap<>();
+    _endpointToClass.put("/kafkacruisecontrol/rebalance", RebalanceParameters.class);
+    _endpointToClass.put("/kafkacruisecontrol/partition_load", PartitionLoadParameters.class);
+  }
+
+  /**
+   * Load the OpenAPI files for endpoints and compares them against the source code
+   */
+  @Test
+  public void loadOpenApiSpec() throws Exception {
+    OpenAPIV3Parser openApiParser = new OpenAPIV3Parser();
+    String baseFileName = "../schemas/base.yaml";
+    ParseOptions options = new ParseOptions();
+    options.setResolve(true);
+    options.setFlatten(true);
+
+    SwaggerParseResult parseResult = openApiParser.readLocation(baseFileName, null, options);
+    _openAPI = parseResult.getOpenAPI();
+    Schema schema = parseSchema();
+    for (Map.Entry<String, Set<String>> endpoint: schema.getEndpointsAndParameters()) {
```

Comment:
Can we add a to do item here to remind us to check the number of endpoint defined in spec is the same as entries in `_endpointToClass`